### PR TITLE
Use strip-bom for removing byte order mark

### PR DESCRIPTION
**Pull request by floatdrop**
Clean up code a little with [strip-bom](https://www.npmjs.com/package/strip-bom).

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -68,18 +69,9 @@ exports.hasProperties = function(obj) {
...
-  .then(function(pjson) {
-    pjson = pjson.toString();
-    // remove any byte order mark
-    if (pjson.startsWith('\uFEFF')) // TODO: https://www.npmjs.com/package/strip-bom
-      pjson = pjson.substr(1);
-    try {
-      return JSON.parse(pjson);
-    }
-    catch(e) {
-      throw 'Error parsing package.json file ' + file;
```
Comment (guybedford) [inline lib/common.js]: This shouldn't be removed.

File: lib/common.js [hunk 2]
```diff
@@ -68,18 +69,9 @@ exports.hasProperties = function(obj) {
 
 exports.readJSON = function(file) {
   return asp(fs.readFile)(file)
-  .then(function(pjson) {
-    pjson = pjson.toString();
-    // remove any byte order mark
-    if (pjson.startsWith('\uFEFF')) // TODO: https://www.npmjs.com/package/strip-bom
-      pjson = pjson.substr(1);
```
Comment (guybedford) [inline lib/common.js]: Why change this?

File: lib/common.js (hunk 1 above)
Comment (floatdrop) [inline lib/common.js]: Yeah, I've added https://github.com/jspm/jspm-cli/pull/572/files#diff-1ffb5a9090758229a335b02a477425c7R78 just after you posted comment.

File: lib/common.js (hunk 2 above)
Comment (floatdrop) [inline lib/common.js]: Just thought, if there is module to do this - why don't use it? It have tests and [works with `Buffer`](https://github.com/sindresorhus/strip-bom/blob/master/index.js#L9-L12) which should be faster a little.

Comment (guybedford) [inline lib/common.js]: I guess I see extra dependency as a cost not a saving. I completely agree for modules that have tests and larger feature sets, but for a one liner I'd rather stick with this.

Comment (floatdrop) [inline lib/common.js]: In some cases this is fair point, but ofter modules with large feature set have huge number of submodules. Besides this extra dependency is loaded once (and it is tiny), when jspm is installed and not affecting bundled package. But this is just my thoughts.

**Comment by guybedford** [comment]
Thanks - will stick with the raw code here for now.

**Comment by floatdrop** [comment]
No problems, sended pr to remove todo from code. Thanks for awesome work you have done!

**Pull request closed without merging**